### PR TITLE
Modernize src/c* src/d* and src/e* files

### PR DESCRIPTION
**Pull request by nerzhul**
* default operator
* redundant init
* delete default constructors on CraftDefinition childs (never used)
* fix some missing init values
* const ref fix reported by clang-tidy
* ranged-based for loops
* simple conditions & returns
* empty stl function instead of size
* emplace_back stl function instead of push_back + construct temp obj
* auto for some iterators
* code style fixes
* c++ stl headers instead of C stl headers (stdio.h -> cstdio)

### CODE REVIEW

File: src/content_cao.cpp [hunk 1]
```diff
@@ -859,13 +856,13 @@ void GenericCAO::addToScene(ITextureSource *tsrc)
 		else
 			errorstream<<"GenericCAO::addToScene(): Could not load mesh "<<m_prop.mesh<<std::endl;
 	}
-	else if(m_prop.visual == "wielditem") {
+	else if (m_prop.visual == "wielditem") {
```
Comment (SmallJoker) [inline src/content_cao.cpp]: Previous bracket belongs into this line.

File: src/content_cao.cpp [hunk 2]
```diff
@@ -1352,12 +1341,11 @@ void GenericCAO::updateTextures(std::string mod)
 				m_meshnode->getMaterial(i).setFlag(video::EMF_ANISOTROPIC_FILTER, use_anisotropic_filter);
 			}
 		}
-		else if(m_prop.visual == "upright_sprite")
-		{
+		else if (m_prop.visual == "upright_sprite") {
```
Comment (SmallJoker) [inline src/content_cao.cpp]: Bracket here too.

File: src/content_cao.cpp [hunk 3]
```diff
@@ -1398,8 +1385,7 @@ void GenericCAO::updateTextures(std::string mod)
 					buf->getMaterial().DiffuseColor = m_prop.colors[1];
 					buf->getMaterial().SpecularColor = m_prop.colors[1];
 				}
-				else if(m_prop.colors.size() >= 1)
-				{
+				else if (!m_prop.colors.empty()) {
```
Comment (SmallJoker) [inline src/content_cao.cpp]: Bracket again

File: src/content_sao.cpp [hunk 4]
```diff
@@ -897,7 +897,7 @@ std::string PlayerSAO::getClientInitializationData(u16 protocol_version)
 	return os.str();
 }
 
-void PlayerSAO::getStaticData(std::string *result) const
+void PlayerSAO::getStaticData(std::string *) const
 {
 	FATAL_ERROR("Deprecated function");
```
Comment (SmallJoker) [inline src/content_sao.cpp]: Would it be possible to remove this function if it's already deprecated?

File: src/daynightratio.h [hunk 5]
```diff
@@ -51,18 +51,19 @@ inline u32 time_to_daynight_ratio(float time_of_day, bool smooth)
 			return values[i][1];
 		}
 		return 1000;
-	} else {
-		for(u32 i=0; i<sizeof(values)/sizeof(*values); i++){
-			if(values[i][0] <= t)
+	}
+
+	for (u32 i=0; i<sizeof(values)/sizeof(*values); i++) {
```
Comment (SmallJoker) [inline src/daynightratio.h]: Space issues.

File: src/content_sao.cpp (hunk 4 above)
Comment (nerzhul) [inline src/content_sao.cpp]: it's a inherited function from ServerActiveObject which should be defined

**Pull request merged**